src/ir/ExpressionManipulator.cpp

```diff
@@ -227,7 +227,9 @@ flexibleCopy(Expression* original, Module& wasm, CustomCopier custom) {
         builder.makeHost(curr->op, curr->nameOperand, std::move(operands));
       return ret;
     }
-    Expression* visitRefNull(RefNull* curr) { return builder.makeRefNull(); }
+    Expression* visitRefNull(RefNull* curr) {
+      return builder.makeRefNull(curr->type);
+    }
```

Comment:
`RefNull` now takes a type, and it uses the provided type's heap type in text and binary format, for instance `(ref.null extern)` making an `externref` of value `null`. Does this look correct?